### PR TITLE
Add browser-based player

### PR DESCRIPTION
**Pull request by spbnick**
This is a hacked-together attempt at implementing a browser-based player, just a starter for discussion for now.

### CODE REVIEW

**Comment by spbnick** [comment]
Alright, this is now on par with tlog-play setup, more-or-less. All controls are implemented. I'll be cleaning this up a bit and submitting a proper PR tomorrow. After this I'll hand it over to you, Kyrill.

**Comment by spbnick** [comment]
@sabbaka, this is now done, please review.

File: pkg/session_recording/player.jsx [hunk 1]
```diff
@@ -0,0 +1,823 @@
...
+         * the packet index is not specified, assume it's the next packet.
+         */
+        awaitPacket(idx) {
+            let i;
+            let idxDfd;
+
+            /* If an error has occurred previously */
+            if (this.error !== null) {
+                /* Reject immediately */
+                return $.Deferred().reject(this.error).promise();
```
Comment (sabbaka) [inline pkg/session_recording/player.jsx:132]: One thing I've found after conversations with Cockpit's team - they are planning to get rid of jQuery, I think we can keep, but maybe we might look forward using underscore.js or something else.
Anyway, our code now is related on jQuery a lot.

File: pkg/session_recording/player.jsx [hunk 2]
```diff
@@ -0,0 +1,823 @@
...
+            } else {
+                /* If it has already been received */
+                if (idx < this.pktList.length) {
+                    /* Return resolved promise */
+                    return $.Deferred().resolve().promise();
+                }
+            }
+
+            /* Try to find an existing, matching tuple */
+            for (i = 0; i < this.idxDfdList.length; i++) {
```
Comment (sabbaka) [inline pkg/session_recording/player.jsx:153]: Tuple here consists from different packets, right?

**Comment by sabbaka** [comment]
Looks good for me, I think I more or less understand what's going on in code.
I have tested it and it works fine.

File: pkg/session_recording/player.jsx (hunk 1 above)
Comment (spbnick) [reply pkg/session_recording/player.jsx:132]: I see. Can we use the [standard Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) instead?

File: pkg/session_recording/player.jsx (hunk 2 above)
Comment (spbnick) [reply pkg/session_recording/player.jsx:153]: A tuple here consists of a packet index and a Deferred object. I described `this.idxDfdList` in the constructor. Would it be better to put the description somewhere else, perhaps?

File: pkg/session_recording/player.jsx (hunk 1 above)
Comment (sabbaka) [reply pkg/session_recording/player.jsx:132]: Yes, since we are using babel and ES6.

File: pkg/session_recording/player.jsx (hunk 2 above)
Comment (sabbaka) [reply pkg/session_recording/player.jsx:153]: Let's leave it as is.

File: pkg/session_recording/player.jsx (hunk 1 above)
Comment (spbnick) [reply pkg/session_recording/player.jsx:132]: Alright, I'll see if I can switch to Promises quickly.

Comment (spbnick) [reply pkg/session_recording/player.jsx:132]: I read about Promises and found out that they don't naturally support the approach I use and switching to them would require some hops and a bit of ugly code. So, I think it is better to leave that for later. Perhaps we'll have a better idea by the time jQuery is going away for real.

Is that OK with you? If yes, I'll merge this as it is.

Comment (sabbaka) [reply pkg/session_recording/player.jsx:132]: Yes, I think it's okay, anyway we are using jQuery a lot.

**Comment by spbnick** [comment]
Thank you, Kyrill, merged!

**Pull request merged**